src/EnergyPlus/ResultsFramework.hh

```diff
@@ -376,78 +371,69 @@ namespace ResultsFramework {
 
         bool MsgPackEnabled() const;
 
-        void initializeRTSDataFrame(const OutputProcessor::ReportingFrequency reportFrequency,
-                                    const Array1D<OutputProcessor::RealVariableType> &RVariableTypes,
-                                    const int NumOfRVariable,
-                                    const OutputProcessor::TimeStepType timeStepType = OutputProcessor::TimeStepType::Zone);
-
-        void initializeITSDataFrame(const OutputProcessor::ReportingFrequency reportFrequency,
-                                    const Array1D<OutputProcessor::IntegerVariableType> &IVariableTypes,
-                                    const int NumOfIVariable,
-                                    const OutputProcessor::TimeStepType timeStepType = OutputProcessor::TimeStepType::Zone);
-
-        void initializeMeters(const Array1D<OutputProcessor::MeterType> &EnergyMeters, const OutputProcessor::ReportingFrequency reportFrequency);
-
-        DataFrame RIDetailedZoneTSData = DataFrame("Detailed-Zone");
-        DataFrame RIDetailedHVACTSData = DataFrame("Detailed-HVAC");
-        DataFrame RITimestepTSData = DataFrame("TimeStep");
-        DataFrame RIHourlyTSData = DataFrame("Hourly");
-        DataFrame RIDailyTSData = DataFrame("Daily");
-        DataFrame RIMonthlyTSData = DataFrame("Monthly");
-        DataFrame RIRunPeriodTSData = DataFrame("RunPeriod");
-        DataFrame RIYearlyTSData = DataFrame("Yearly");
-        MeterDataFrame TSMeters = MeterDataFrame("TimeStep");
-        MeterDataFrame HRMeters = MeterDataFrame("Hourly");
-        MeterDataFrame DYMeters = MeterDataFrame("Daily");
-        MeterDataFrame MNMeters = MeterDataFrame("Monthly");
-        MeterDataFrame SMMeters = MeterDataFrame("RunPeriod");
-        MeterDataFrame YRMeters = MeterDataFrame("Yearly");
+        void initializeTSDataFrame(const ReportFreq reportFrequency,
+                                   const std::vector<OutputProcessor::OutVar*> &Variables,
+                                   const OutputProcessor::TimeStepType timeStepType = OutputProcessor::TimeStepType::Zone);
+            
+        void initializeMeters(const std::vector<OutputProcessor::Meter*> &EnergyMeters, const ReportFreq reportFrequency);
+
+        std::array<DataFrame, (int)TimeStepType::Num> detailedTSData = {
```

Comment:
Keep these data frames in arrays so we can access them by indexing with reportFreq.